onedal/utils/validation.py

```diff
@@ -438,3 +443,31 @@ def _is_csr(x):
     return isinstance(x, sp.csr_matrix) or (
         hasattr(sp, "csr_array") and isinstance(x, sp.csr_array)
     )
+
+
+def _assert_all_finite(X, allow_nan=False, input_name=""):
+    policy = _get_policy(None, X)
+    params = {
+        "fptype": "float" if X.dtype.name == "float32" else "double",
+        "method": "dense",
+        "allow_nan": allow_nan,
+    }
+    X_t = to_table(_convert_to_supported(policy, X))
+    if not _backend.finiteness_checker.compute.compute(policy, params, X_t).finite:
+        type_err = "infinity" if allow_nan else "NaN, infinity"
+        padded_input_name = input_name + " " if input_name else ""
+        msg_err = f"Input {padded_input_name}contains {type_err}."
+        raise ValueError(msg_err)
+
+
+def assert_all_finite(
+    X,
+    *,
+    allow_nan=False,
+    input_name="",
+):
+    _assert_all_finite(
+        X.data if sp.issparse(X) else X,
+        allow_nan=allow_nan,
+        input_name=input_name,
+    )
```

Comment:
Is this main API for the feature? Is it used in the tests?

Reply:
Added test which uses the main function name, also testing sparse arrays as well.